docs/ir/ir-breaking-changes.rst

```diff
@@ -132,4 +172,35 @@ the right function.
 The ID ``0`` is reserved for uninitialized function pointers which then cause a panic in the disptach function when called.
 
 In the old code generator, internal function pointers are initialized with a special function that always causes a panic.
-This causes a storage write at construction time for internal function pointers in storage.
\ No newline at end of file
+This causes a storage write at construction time for internal function pointers in storage.
+
+Clean up
+--------
+
+.. index:: cleanup, dirty bits
+
+The old code generator only performs cleanup before an operation that could be affected by the the values of the dirty bits.
```

Comment:
```suggestion
The old code generator only performs cleanup before an operation whose result that could be affected by the the values of the dirty bits.
```
Maybe also more explicitly: The old generator delays cleanup if the cleaned result of an operation is the same regardless of whether the inputs are cleaned or not.